volunteer.php

```diff
@@ -650,4 +664,62 @@ function volunteer_civicrm_fieldOptions($entity, $field, &$options, $params) {
   if ($entity == "UFJoin" && $field == "entity_table" && $params['context'] == "validate") {
     $options[CRM_Volunteer_DAO_Project::getTableName()] = CRM_Volunteer_DAO_Project::getTableName();
   }
+}
+
+/**
+ * This function fetches the defaults from civicrm settings
+ * And puts them into the appropriate data format to return
+ * to the angular front-end
+ *
+ * @return array
+ * @throws CiviCRM_API3_Exception
+ */
+function volunteer_composeDefaultArray() {
```

Comment:
It is not specific to the API, and I do foresee it being used elsewhere in future. It really should go in the ProjectUtil BAO, which doesn't exist. Perhaps I should move it to the Project BAO, but I don't think it belongs in the API